PSI_Tracker.js

```diff
@@ -0,0 +1,645 @@
+    let last_column = sheet.getLastColumn();
+    let values = sheet.getRange(1, 1, last_row, last_column).getValues();
+    // Go line by line of all URLs and change value and note
+    for (let i = 1; i < values.length; i++) {
+        const url = values[i][4];
+        const active = values[i][7]; //G
+        sheet.getRange(i+1,COLUMN_STATUS+1).setValue("")
+        sheet.getRange(i+1,COLUMN_STATUS+1).setNote("")
+        // Skip empty lines
+        if(url == "" || url == null || url == undefined){continue;}
+        if(!active){continue;}
+        sheet.getRange(i+1,COLUMN_STATUS+1).setValue("⏳")
+    }
+    SpreadsheetApp.flush();
+}
+
+/**
+ * Run Batch, reads the number of URLs to send to the API and runs a batch of them
+ */
+function runBatch(api = "PSI", save_screenshot = false){
+    let sheet = CONFIG_SHEET;
+    let NUMBER_OF_URLS_PER_BATCH = +sheet.getRange("B21").getValue();
+    let last_row = sheet.getLastRow();
+    let last_column = sheet.getLastColumn();
+    let values = sheet.getRange(1, 1, last_row, last_column).getValues();
+
+    // STEP 1: Create a batch of NUMBER_OF_URLS_PER_BATCH URL
+    let batch = [];
+    for (let i = 1; i < values.length; i++) {
+
+        // Read the values from column D to H
+        const label = values[i][3];  //D
+        const url = values[i][4];    //E
+        const device = values[i][5]; //F
+        const origin = values[i][6]; //G
+        const active = values[i][7]; //G
+        const status = values[i][COLUMN_STATUS]; //H
+        
+        // Skip empty content
+        if(url == "" || url == null || url == undefined){continue;}
+        if(!active){continue;}
+
+        // Add the line to the current batch
+        if(status == "⏳"){
+            // Mark the status as being worked on
+            sheet.getRange(i+1,COLUMN_STATUS+1).setValue("🔃")
+            // Check if "Mobile" or "Phone" was requested
+            if(device.toLowerCase().indexOf("mobile") != -1 || device.toLowerCase().indexOf("phone") != -1){
+                batch.push({id:i, label:label, url:url, device:"MOBILE", origin:origin});
+            }
+            // Check if "Desktop" was requested
+            if(device.toLowerCase().indexOf("desktop") != -1){
+                batch.push({id:i, label:label, url:url, device:"DESKTOP", origin:origin});
+            }
+            // Break if we went over the number of maximum URLs per batch (±1)
+            if(batch.length >= NUMBER_OF_URLS_PER_BATCH){break;}
+        }
+    }
+
+    // STEP 2: Send the batch to the API
+    toast("Batch","Sending " + JSON.stringify(batch));
+    // Use newPSIRequest to make GET requests to the API, for the PSI API they look like this:
+    // https://googleapis.com/pagespeedonline/v5/runPagespeed?key=API_KEY&category=ACCESSIBILITY&category=BEST_PRACTICES&category=PERFORMANCE&category=PWA&category=SEO&strategy=MOBILE&url=http://web.dev
+    let urls = [];
+    if(api == "PSI"){urls = batch.map(d=>(newPSIRequest(d.url,d.device)))}
```

Comment:
If you used a switch you could combine some to show that the default is a PSI request.

Reply:
makes sense, I found it relatively readable as is, happy to switch to switch!

Reply:
A switch with a break is slightly more efficient, or you should use `else if` so that you're not checking the value of api 6 times every time.